src/components/TsmlUI.tsx

```diff
@@ -100,138 +108,143 @@ export default function TsmlUI({
     };
   }, []);
 
-  // load data once
-  if (state.loading) {
-    console.log(
-      'TSML UI meeting finder: https://github.com/code4recovery/tsml-ui'
-    );
-
-    const input = getQueryString(settings);
-
-    if (!src) {
-      setState({
-        ...state,
-        error: 'Configuration error: a data source must be specified.',
-        loading: false,
-        ready: true,
-      });
-    } else {
-      const sheetId = src.startsWith('https://docs.google.com/spreadsheets/d/')
-        ? src.split('/')[5]
-        : undefined;
-
-      // google sheet
-      if (sheetId) {
-        if (!google) {
-          setState({
-            ...state,
-            error: 'Configuration error: a Google API key is required.',
-            loading: false,
-          });
-        }
-        src = `https://sheets.googleapis.com/v4/spreadsheets/${sheetId}/values/A1:ZZ?key=${google}`;
-      }
+  useEffect(() => {
```

Comment:
wild that this was never in a useEffect - can probably be combined with the one above. don't need to check for `state.loading` now either